python/cuml/benchmark/algorithms.py

```diff
@@ -152,18 +176,20 @@ def all_algorithms():
         AlgorithmPair(
             sklearn.random_projection.GaussianRandomProjection,
             cuml.random_projection.GaussianRandomProjection,
-            shared_args=dict(n_components=10),
+            shared_args=dict(n_components="auto"),
             name="GaussianRandomProjection",
+            bench_func=fit_transform,
             accepts_labels=False,
         ),
         AlgorithmPair(
             sklearn.neighbors.NearestNeighbors,
             cuml.neighbors.NearestNeighbors,
             shared_args=dict(n_neighbors=1024),
-            cpu_args=dict(algorithm="brute"),
+            cpu_args=dict(algorithm="brute", n_jobs=-1),
```

Comment:
I have this theory that maybe we should set n_jobs globally somehow so it threads through any sklearn method? I find I often want to pin a specific n_jobs to get consistency across machines or to test single-cpu... it's ok to add your change here now, just starting to think that one through...

Reply:
Hmm. Given the current design, we have a few options. I'd prefer not to set it globally, but maybe add it as a parameter on the runner that can be automatically propagated to any algorithm pairs it runs. I think if we add this to the constructor, rather than the run function, it would reduce the code even further. We could:

1. Set an explicit argument in the constructor of the runner, and maybe call it `cpu_n_jobs`
2. Provide an argument in the constructor for general cpu/gpu params, which get merged in with the other params (original or overrides).

Do you have any other ideas? I'm more partial to #2 just because I think it sets us up for things like this in the future without having to make code changes each time (and also enabling future users to configure it to their needs).

Reply:
I like #2, but it has to be smart enough that it doesn't crash if passed to an algo that lacks the n_jobs parameter (I believe a few are missing this).

Reply:
Okay. I might do this one in a different PR (probably aimed at 0.11) so I can play with a reasonable way to do this.

Reply:
Created #1229 for this.